spec/services/api/declarations/query_spec.rb

```diff
@@ -0,0 +1,475 @@
+          let(:ect_at_school_period) { training_period1.ect_at_school_period }
+
+          # Current training period
+          let(:prev_finished_on) { training_period1.started_on + rand(100).days }
+          let(:training_period2) do
+            # set previous training period to end
+            training_period1.update!(finished_on: prev_finished_on)
+
+            FactoryBot.create(:training_period, :for_ect, ect_at_school_period:, started_on: prev_finished_on, school_partnership: school_partnership2)
+          end
+
+          # New training period
+          let(:curr_finished_on) { training_period2.started_on + rand(100).days }
+          let(:training_period3) do
+            # set current training period to end
+            training_period2.update!(finished_on: curr_finished_on)
+
+            FactoryBot.create(:training_period, :for_ect, ect_at_school_period:, started_on: curr_finished_on, school_partnership: school_partnership3)
+          end
+
+          # Declaration for each training period
+          let!(:declaration1) do
+            FactoryBot.create(
+              :declaration,
+              training_period: training_period1,
+              declaration_type: "started",
+              declaration_date: training_period1.started_on.next_day
+            )
+          end
+          let!(:declaration2) do
+            FactoryBot.create(
+              :declaration,
+              training_period: training_period2,
+              declaration_type: "retained-1",
+              declaration_date: training_period2.started_on.next_day
+            )
+          end
+          let!(:declaration3) do
+            FactoryBot.create(
+              :declaration,
+              training_period: training_period3,
+              declaration_type: "retained-2",
+              declaration_date: training_period3.started_on.next_day
+            )
+          end
+          # Additional unrelated declaration
+          let!(:declaration4) { FactoryBot.create(:declaration) }
+
+          context "when lead_provider2 has previous and direct declaration" do
+            it "returns previous declarations and direct declarations for lead_provider2" do
+              query = described_class.new(lead_provider_id: lead_provider2.id)
+
+              expect(query.declarations).to contain_exactly(declaration1, declaration2)
+            end
+          end
+
+          context "when previous declaration is submitted after training_period2.finished_on date" do
+            before { declaration1.update!(declaration_date: training_period2.finished_on.next_day) }
+
+            it "returns only the direct declaration" do
+              query = described_class.new(lead_provider_id: lead_provider2.id)
+
+              expect(query.declarations).to contain_exactly(declaration2)
+            end
+          end
+
+          context "when previous declaration is voided" do
```

Comment:
It might be worth wrapping this in a loop to also test the other states as well?:

```
%i[ineligible voided clawback clawed_back].each do |ignored_status|
  context "when previous declaration is #{ignored_status}" do
    ...
  end
end
```